src/core/main.js

```diff
@@ -231,11 +231,18 @@ class p5 {
       this._events.devicemotion = null;
     }
 
-    // before and after hooks for preload and setup
-    this._beforePreload = function () {};
-    this._afterPreload = function () {};
-    this._beforeSetup = function () {};
-    this._afterSetup = function () {};
+    // Function to call registered before and after hooks for preload and setup
+    function callRegisteredHooksFor(hookName) {
+      const target = this || p5.prototype;
+      if (target._registeredMethods.hasOwnProperty(hookName)) {
+        const methods = target._registeredMethods[hookName];
+        for (const method of methods) {
+          if (typeof method === 'function') {
+            method.call(this);
```

Comment:
Nice work on this helper function! Just for consistency, now that we have this helper, can we update the pre/post draw hooks to use this too? Currently that happens here: https://github.com/processing/p5.js/blob/63f94b035a9361f3846f9cd3cc8f839666766628/src/core/structure.js#L486-L486

I also notice that for pre/post draw, rather than passing `this` to `call`, they pass `context`, where it is defined as: https://github.com/processing/p5.js/blob/63f94b035a9361f3846f9cd3cc8f839666766628/src/core/structure.js#L469
Maybe we can do that here too?